packages/redux-window-message-thunk/src/redux-window-message-thunk.ts

```diff
@@ -0,0 +1,28 @@
+import { Action, Dispatch, Middleware } from 'redux';
+
+interface MessageHandler {
+  [key: string]: (message: MessageEvent) => Action
+}
+
+export const messageListener = async (domains: string[], messageHandlers: MessageHandler, dispatch: Dispatch, message: MessageEvent) => {
+  const { data, origin } = message;
+  const { type } = data;
```

Comment:
@itaylor Based on your comments regarding flattening the data object sent from the iframe, I've updated this so the messageCallback is passed the entire data object and no longer nested under payload.